tests/test_utils/test_colo_checkpoint.py

```diff
@@ -155,43 +159,33 @@ def _run_checkpoint(model_name, init_spec_func, use_ddp, use_mp_reload, test_sch
         loss.backward()
         loss_reload.backward()
 
-        if run_reload:
-            colo_optimizer_reload.zero_grad()
-            if criterion:
-                output_reload = model_reload(data)
-                loss_reload = criterion(output_reload, label)
-            else:
-                loss_reload = model_reload(data, label)
-            loss_reload.backward()
-            colo_optimizer_reload.step()
+        colo_optimizer.step()
+        colo_optimizer_reload.step()
 
         if i > 2:
             break
 
     if not os.path.isdir('./checkpoint') and rank == 0:
         os.mkdir('./checkpoint')
-    save_checkpoint('./checkpoint', 0, model, colo_optimizer, None)
-    dist.barrier()
-    load_checkpoint('./checkpoint', 0, model_reload, colo_optimizer_reload, None)
     dist.barrier()
 
-    # Since model is sharded, we merge them before param checking.
-    for p in model.parameters():
-        p.to_replicate_()
-
-    for p in model_reload.parameters():
-        p.to_replicate_()
+    save_checkpoint('./checkpoint', 0, model, colo_optimizer, None)
+    load_checkpoint('./checkpoint', 0, model_reload, colo_optimizer_reload, None)
 
     check_param_equal(model, model_reload)
     compare_optims(colo_optimizer, colo_optimizer_reload)
+
     if rank == 0:
         remove('./checkpoint')
+    dist.barrier()
 
 
 def run_dist(rank, world_size, port, use_ddp, use_mp_reload, test_scheduler):
     colossalai.launch(config={}, rank=rank, world_size=world_size, host='localhost', port=port, backend='nccl')
     pg = ProcessGroup(tp_degree=world_size)
-    for model_name in ['simple_net', 'bert']:
+    # TODO(haichen) add BERT in the test
```

Comment:
Inside a DP group, the input is replicated?

Reply:
Depends on which model is using, we do not unite the standard now.